tests/unit/test_service.py

```diff
@@ -38,3 +51,112 @@ def test_install_apt_package_gpg_key_error(
 
     with pytest.raises(RuntimeError, match="Error installing the agent service"):
         charm._on_install(MagicMock(spec=ops.InstallEvent))
+
+
+def test_on_install(harness: ops.testing.Harness, monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: Harness with mocked apt module.
+    act: run initial hook.
+    assert: The installation should pass without error and charm in blocked state.
+    """
+    apt_repository_mapping_mock = MagicMock()
+    apt_import_key_mock = MagicMock()
+    apt_update_mock = MagicMock()
+    apt_add_package_mock = MagicMock()
+
+    apt_repository_mapping_mock.__contains__.return_value = False
+    monkeypatch.setattr(apt, "RepositoryMapping", apt_repository_mapping_mock)
```

Comment:
You should probably aim to mock here only the lib methods you are directly calling. It should simplify the mocking and abstract the test from the underlying lib implementation, which can cause the tests to break even when the methods you are using are unchanged

Reply:
I actually called those lib methods to be able to install the agent's apt package: https://github.com/canonical/jenkins-agent-operator/blob/7c5ccfd8829b72abe2f5a9079ecd3f99c31192c6/src/service.py#L105

In a harness environment those calls will fail. Do you think we should abstract away those lib calls into a separate so that we only need to mock the abstraction during tests? But the abstraction will still count towards the coverage target.

Reply:
Makes sense. Can you assert that the `apt_add_package_mock` is actually called with the expected packages as args?

Reply:
Yes, I added checks for the call to install of the correct package